packages/core/src/main/conversation/ConversationService.ts

```diff
@@ -92,13 +92,38 @@ import type {
   ResetSessionMessage,
   TextMessage,
 } from './message/OtrMessage';
+import {XOR} from '@wireapp/commons/src/main/util/TypeUtil';
 
 export enum MessageTargetMode {
   NONE,
   USERS,
   USERS_CLIENTS,
 }
 
+interface SendCommonParams<T> {
+  protocol: ConversationProtocol;
+  payload: T;
+  onStart?: (message: GenericMessage) => void | boolean | Promise<boolean>;
+  onSuccess?: (message: GenericMessage, sentTime?: string) => void;
+  onClientMismatch?: (
+    status: ClientMismatch | MessageSendingStatus,
+    wasSent: boolean,
+  ) => void | boolean | Promise<boolean>;
```

Comment:
This is only for Proteus messages, MLS messages won't ever have mismatch (eventually epoch failure but that's a completely different process)